src/Models/SubscriptionProduct.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Payavel\Subscription\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class SubscriptionProduct extends Model
+{
+    /**
+     * The attributes that aren't mass assignable.
+     *
+     * @var string[]|bool
+     */
+    protected $guarded = ['id'];
+
+    /**
+     * Get the Subscriptions' Product
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function subscriptions()
+    {
+        return $this->hasMany(config('subscription.models.' . Subscription::class, Subscription::class));
+    }
+
+    /**
+     * Get the Plans' Product
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function plans()
+    {
+        return $this->hasMany(config('subscription.models.' . SubscriptionPlan::class, SubscriptionPlan::class));
```

Comment:
No need to look at the config, let's go straight for the `SubscriptionPlan::class` model.